src/controllers/Role.controller.ts

```diff
@@ -0,0 +1,132 @@
+import { Request, Response } from "express";
+import User from "../database/models/User.model";
+import { RoleAndPermissionAttribute } from "../interfaces/Roles.interface";
+import {
+  deleteRoleService,
+  getAllRolesService,
+  setRoleServise,
+} from "../services/Role.service";
+import { isValidUuid } from "../utils";
+
+// Get roles
+export const allRoles = async (req: Request, res: Response) => {
+  try {
+    const allRolesAndPermissionsInDB = await getAllRolesService();
+    res.status(200).json({
+      status: 200,
+      success: true,
+      data: allRolesAndPermissionsInDB,
+    });
+  } catch (error) {
+    if (error instanceof Error) {
+      return res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error occur while getting all roles",
+        error: error.message,
+      });
+    }
+  }
+};
+
+// Admin set roles in db
+export const setRole = async (req: Request, res: Response) => {
+  try {
+    const role: RoleAndPermissionAttribute = req.body;
+
+    const settedRole = await setRoleServise(role);
+
+    return res.status(200).json({
+      status: 200,
+      success: true,
+      message: "Role setting is done successfully!",
+      data: settedRole,
+    });
+  } catch (error) {
+    if (error instanceof Error) {
+      return res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error occur while seting role in db",
+        error: error.message,
+      });
+    }
+  }
+};
+
+// Admin set role to a particular user
+export const setRoleToUser = async (req: Request, res: Response) => {
+  try {
+    const userId = req.params.id;
+
+    if (isValidUuid(req.body.roleId)) {
+      const updatedUserRole = await User.update(
+        { roleId: req.body.roleId },
+        {
+          where: { id: userId },
+          returning: true,
+        },
+      );
+
+      return res.status(200).json({
+        status: 200,
+        success: true,
+        data: updatedUserRole,
+      });
+    } else {
+      return res.status(400).json({
+        status: 400,
+        success: false,
+        message: "Please provide a valid uuid for roleId",
+      });
+    }
+  } catch (error) {
+    if (error instanceof Error) {
+      return res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error occur while giving role a specified user",
+        error: error.message,
+      });
+    }
+  }
+};
```

Comment:
@ISHIMWE-Aime refactor this code block

Reply:
I am sorry but I am not getting clearly what you mean here.
Thank you.